### PR TITLE
The Grand French Localization Update

### PR DESCRIPTION
**Pull request by Setnour6**
**I need official French linguists to help me confirm translations, because my basic French does not cut it.**

### What did you just do?
I updated the French Translations and updated every single entry.
### Why should this be part of tModLoader?
The French Translations and Localizations were incomplete, so I took the initiative to actually finish it using my basic French knowledge and two translators (Google and Yandex)
### Are there alternative designs?
_no response_
### Sample usage for the new feature
_no response_
### ExampleMod updates

_no response_

### CODE REVIEW

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 1]
```diff
@@ -3,22 +3,22 @@
 		// Main Menu
 		"PatreonSupport": "Considerez de nous supporter financièrement: ",
 		"MenuModBrowser": "Navigateur",
-		// "MenuDownloadMods": "Download Mods",
-		// "MenuDownloadModsDescription": "The in-game Mod Browser to find, download, and update your Mods.",
+		// "MenuDownloadMods": "Télécharger des Mods",
+		// "MenuDownloadModsDescription": "Le Bavigateur de Mods dans le jeu pour trouver, télécharger, et mettre à jour votre Mods.",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Minor typo here, it's "Navigateur" not Bavigateur, a comma just before an "et" is usually discouraged and "votre" should be in plural ; "vos"

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 2]
```diff
@@ -3,22 +3,22 @@
...
 		"MenuMods": "Mods",
-		// "MenuManageMods": "Manage Mods",
-		// "MenuManageModsDescription": "Manage which Mods are active, delete Mods, and configure your Modpacks.",
-		"MenuModSources": "Mods source",
-		// "MenuDevelopMods": "Develop Mods",
-		// "MenuDevelopModsDescription": "Develop and publish Mods for others to use, using your Mod Source code",
+		// "MenuManageMods": "Gérer les Mods",
+		// "MenuManageModsDescription": "Gérez quels Mods sont actifs, supprimez les mods, et configurez votre Modpacks.",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: In "supprimez les mods", I believe it should be  "supprimez des Mods"
Also, modpacks is a plural so it hould be "vos Modpacks."

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
+		"MenuModSources": "Sources de Mods",
+		// "MenuDevelopMods": "Développer des Mods",
+		// "MenuDevelopModsDescription": "Développer et publier des Mods pour que d’autres puissent les utiliser, en utilisant votre Code Source Mod",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: "Code Source _de_ Mod" looks better

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 4]
```diff
@@ -3,22 +3,22 @@
...
-		// "FirstLaunchWelcomeMessage": "Welcome to tModLoader.\n\nYou have successfully installed tModLoader. Use the Mod Browser menu to find mods you wish to download. Use the Mods menu to change which mods are currently active. After changing mods, make sure you click the Reload Mods button to reload them.",
-		// "WhatsNewMessage": "The following are basic change notes from the last time you launched the Alpha. You may find that mods that relate to these changes need to be rebuilt by the author. Visit GitHub to see more details:\n",
-		// "ViewOnGitHub": "View on GitHub",
-		// "tModLoaderSettings": "tModLoader Settings",
-		// "ModMenuSwap": "Switch Menu Theme",
-		// "AudioNotSupported": "No audio hardware found. Disabling all audio.",
+		// "WelcomeMessageBeta": "Bienvenue sur tModLoader 1.4 Alpha.\n\nVous avez installé avec succès une version instable de tModLoader qui peut contenir des bogues de rupture de jeu. \n\nUtilisez le menu Atelier pour accéder aux fonctions liées à tModLoader Mod.\n\nRejoignez le Discord tModLoader pour en savoir plus sur les mods et/ou outils de test disponibles pour tModLoader 1.4 dans le cadre de sa refonte 1.4.",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Honestly, no one use "bogues" and nearly everyone talks about "bugs" in French, even if that's not what dictionaries say.
I'd understand to leave it as "bogue" though

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 5]
```diff
@@ -29,18 +29,18 @@
...
-		// "ModsOpenConfig": "Click to open config",
+		// "ModConfiguration": "Configuration du Mod",
+		// "ModsOpenConfig": "Cliquez pour ouvrir la configuration",
 		"ModOldWarning": "Mod obsolète, activer à vos risques et périls",
 		"ModDependencyClickTooltip": "Ce Mod dépend de : {0}\n (Cliquer pour activer)",
 		"ModDependencyModsNotFound": "Ces Mods n'ont pas été trouvés: {0}",
-		// "ModDidNotFullyUnloadWarning": "This mod did not fully unload during last unload.\n(This is a message for the mod creator)",
-		"BetaModCantPublish": "Ce Mod a été compilé en version Beta et ne peut pas être publié",
+		// "ModDidNotFullyUnloadWarning": "Ce mod ne s'est pas complètement déchargé lors du dernier déchargement.\n(Ceci est un message pour le créateur du mod)",
+		"BetaModCantPublish": "Ce mod a été construit sur la version bêta et ne peut pas être publié",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: The old version of "BetaModCantPublish" was very nice to me, with correct uppercasing of Mod and "Beta" should stay "Beta". "bêta" is someone stupid in French

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 6]
```diff
@@ -53,43 +53,43 @@
 		"ModsShowAllMods": "Tout montrer",
 		"ModsShowEnabledMods": "Montrer uniquement les Mods actifs",
 		"ModsShowDisabledMods": "Montrer uniquement les Mods inactifs",
-		// "ModsXModsFilteredByEnabled": "{0} {^0:mod;mods} filtered by enabled filter.",
-		// "ModsXModsFilteredByModSide": "{0} {^0:mod;mods} filtered by enabled mod side filter.",
-		// "ModsXModsFilteredBySearch": "{0} {^0:mod;mods} filtered by enabled search filter.",
+		// "ModsXModsFilteredByEnabled": "{0} {^0:mod;mods} filtré par filtre activé.",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: "par le filtre actif" would look better I think

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 7]
```diff
@@ -53,43 +53,43 @@
...
 		"ModsXTiles": "{0} {^0:block;blocks}",
 		"ModsXWalls": "{0} {^0:mur;murs}",
 		"ModsXBuffs": "{0} {^0:buff;buffs}",
 		"ModsXMounts": "{0} {^0:monture;montures}",
-		// "ModsReloadAndReturn": "Reload and return to world menu",
-		// "ModsNotFoundServer": "No mods were found in: \"{0}\"\nIf you are running a dedicated server, you may wish to use the 'modpath' command line switch or server config setting to specify a custom mods directory.\n",
-		// "AskForModIndex": "Type a number to switch between enabled/disabled",
-		// "AskForCommand": "Type a command: ",
+		// "ModsReloadAndReturn": "Recharger et retourner au menu du monde",
+		// "ModsNotFoundServer": "No mods were found in: \"{0}\"\nSi vous exécutez un serveur dédié, vous pouvez utiliser le commutateur de ligne de commande 'modpath' ou le paramètre de configuration du serveur pour spécifier un répertoire de mods personnalisés.\n",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Shouldn't  "No mods were found in: \" translated by "Aucun Mod n'a été trouvé à :\" ?

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json [hunk 8]
```diff
@@ -375,77 +375,77 @@
...
-		// "InLihzardTemple": "Needs to be in Lihzard Temple",
-		// "InGraveyardBiome": "Needs to be in Graveyard Biome"
+		// "NearWater": "Besoins d’eau",
+		// "NearLava": "Besoins de lave",
+		// "NearHoney": "Besoins de miel",
+		// "TimeDay": "Besoins de jour",
+		// "TimeNight": "Besoin de nuit",
+		// "InDungeon": "Doit être dans le Biome du Donjon",
+		// "InCorrupt": "Doit être dans le biome de la Corruption",
+		// "InHallow": "Doit être dans le biome de la Sacré",
```
Comment (Ishigh1) [inline patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Why is "biome" sometimes with and sometimes without an uppercase ?

**Review by Ishigh1 — COMMENTED** [review]
I'm French so even without being an official French Linguist I can give my opinion about few things (Some things appeared more than once but I left a comment / type of "issue")

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json (hunk 4 above)
Comment (Setnour6) [reply patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: I can use "bugs" if you want. For now, I put a note.

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json (hunk 6 above)
Comment (Setnour6) [reply patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: I assume this applies to the other two lines of mod filter...?

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json (hunk 7 above)
Comment (Setnour6) [reply patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: I just realized I missed that. Thank you for noticing

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json (hunk 8 above)
Comment (Setnour6) [reply patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Whoops, that wasn't supposed to happen. I just fixed it.

File: patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json (hunk 6 above)
Comment (Ishigh1) [reply patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json]: Yeah, didn't write the same thing twice on this review

**Comment by Setnour6** [comment]
I'm not sure if you got notified about the new commits but I made the fixes

**Comment by Solxanich** [comment]
PLease use a branch for pull requests - the commit history on this one contains irrelevant commits, making it difficult to consider merge.
Can you please migrate the changes to a fresh branch? 

Steps to do so:
1) Take a copy of the file patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json , place it somewhere outside of the dev workspace for now
2) Create a branch off of upstream/1.4. Call it whatever, maybe 1.4_french
3) Pull the branch & regenerate source in the setup tool
4) The the copied file from step 1, and replace the new patches/tModLoader/Terraria/Localization/Content/fr-FR/tModLoader.json
5) Run the diff in setup tool again
6) push, create new PR, & profit

**Comment by Setnour6** [comment]
I had so much trouble trying to regenerate the source, mostly because I don't know how it works and what it means to me. I only found setup.bat, but I kept getting child -1 - forked process {Number} died unexpectedly.
I deleted the fork, so I'll fork this repository again and work from there.

**Pull request closed without merging**